custom/icds_reports/utils/aggregation_helpers/tests/sql_output/growth-monitoring-forms.distributed.txt

```diff
@@ -13,133 +13,306 @@ DELETE FROM "icds_dashboard_growth_monitoring_forms" WHERE month=%(month)s AND s
             muac_grading, muac_grading_last_recorded
         ) (
           SELECT
-            %(state_id)s AS state_id,
-            COALESCE(ucr.supervisor_id, prev_month.supervisor_id) AS supervisor_id,
+            state_id, supervisor_id, %(month)s, case_id, latest_time_end_processed,
+            weight_child, weight_child_last_recorded,
+            height_child, height_child_last_recorded,
+            zscore_grading_wfa, zscore_grading_wfa_last_recorded,
+            zscore_grading_hfa, zscore_grading_hfa_last_recorded,
+            zscore_grading_wfh, zscore_grading_wfh_last_recorded,
+            muac_grading, muac_grading_last_recorded
+          FROM "icds_dashboard_growth_monitoring_forms"
+          WHERE month = %(previous_month)s AND state_id = %(state_id)s
+      )
+        
+{"month": "2019-01-01", "previous_month": "2018-12-01", "state_id": "st1"}
+
+        INSERT INTO "icds_dashboard_growth_monitoring_forms" as gm_forms (
```

Comment:
All these insert queries look very much alike. Could another level of abstraction be applied here. Something like:
create a function `prepare_insert_query` similar the way you made `data_from_ucr_query`. and calling `prepare_insert_query` in loop with diffrent param and yielding the query.